src/lcm/simulate.py

```diff
@@ -655,32 +575,6 @@ def dict_product(d):
     return dict(zip(d.keys(), list(stacked.T), strict=True)), len(stacked)
 
 
-def create_choice_segments(mask, n_sparse_states):
-    """Create choice segment info related to sparse states and choices.
-
-    Comment: Can be made more memory efficient by reshaping mask into 2d.
-
-    Args:
-        mask (jnp.ndarray): Boolean 1d array, where each entry corresponds to a
-            data-state-choice combination.
-        n_sparse_states (np.ndarray): Number of sparse states
-
-    Returns:
-        dict: Dict with segment_info.
-
-    """
-    n_choice_combinations = len(mask) // n_sparse_states
-    state_ids = jnp.repeat(
-        jnp.arange(n_sparse_states),
-        repeats=n_choice_combinations,
-    )
-    segments = state_ids[mask]
-    return {
-        "segment_ids": jnp.array(segments),
-        "num_segments": len(jnp.unique(segments)),
-    }
-
-
 def determine_discrete_dense_choice_axes(variable_info):
```

Comment:
Seems the same as `discrete_problem._determine_dense_discrete_choice_axes`

Reply:
No they are not the same, but it is not clear at all from the code why; and that needs to change!

The condition in the first query is different and the choice indices in the function in the simulation module need to be offset by 1, to account for the initial states (sparse state variables).

I have added it to the list of ToDo's for the refactoring.